validator/db/kv/proposal_history_v2.go

```diff
@@ -54,6 +59,13 @@ func (store *Store) SaveProposalHistoryForPubKeysV2(
 				return fmt.Errorf("could not create bucket for public key %#x", pubKey)
 			}
 			for _, proposal := range history.Proposals {
+				minimalSlot, ok := minimalProposalSlot[pubKey]
```

Comment:
it would be smarter to save the minimal slot after all your proposals. Otherwise you risk saving the minimal slot after looping through each proposal